### ISSUE TITLE
Incorrect effective positioner angles

### ISSUE THREAD

**Issue by tskisner** [desihub/fiberassign#266]
This issue is to track a critical problem discovered as part of recent effort to understand fiberassign efficiencies.  Thanks to Anand and @Cyeche for help in illustrating the issue.  

It looks like the theta / phi angle ranges of each positioner are not being referenced or rotated correctly to the petal they are in.  This results in an effectively reduced range of the positioners as the petals go from location 3 (at zero rotation from the coordinate system) around to location 2.

This can be illustrated with a small test on one tile that samples a high uniform density of targets (e.g. 50000 / sq deg), computes the targets available to all positioners, and then plots the targets **not** available to **any** positioner.  See attached plots of the positioner locations and the "unreachable" targets.  Petal location 3 has the expected spots where targets might be unreachable, while petal location 2 has many more in a suspicious angular slice of the patrol region.

This *might* be a desimodel issue rather than fiberassign, but I will leave this issue here until it is clear the source of the problem.

![coverage_50000](https://user-images.githubusercontent.com/84221/81142920-b1eeca80-8f25-11ea-8712-0fefed14aa84.png)
![target_reachability_zoom](https://user-images.githubusercontent.com/84221/81142947-ba470580-8f25-11ea-8c8e-e5e11bb945c3.png)

**Comment by sbailey** [comment]
Does fiber assign treat the theta/phi limits in a global focal plane coordinate system (e.g. CS5), or petal local?  The online calibs use petal-local, and presumably the calibs we inherited from the teststand do too.  I don't immediately see how that difference would cause this effect, but it might be related and is something that we do need to be consistent about.

**Comment by forero** [comment]
This looks like the same pattern on the issue about sky fibers

https://github.com/desihub/fiberassign/issues/151#issuecomment-604804294

The pattern disappears by running with at date `2019-01-01T00:00:00`

https://github.com/desihub/fiberassign/issues/151#issuecomment-606953386

**Comment by tskisner** [comment]
Thanks @forero , that is a good data point.  Any dates before 2019-09-16 will use the legacy focalplane from the old fiberassign (but with values put into the desimodel format and retrieved from there).  I'll try to track this down later this evening.

**Comment by tskisner** [comment]
Ok, this turned out to be an unfortunate combination of things.  The fix was relatively easy, but I'm now working on a unit test for this situation before opening a PR.  Here I want to document the sequence of events for the record.

1.  The legacy fiberassign had hard-coded values for the positioner range of motion and arm lengths.

2.  When moving these hardcoded values into the desimodel focalplane framework, the scripts to generate focalplane models took various options that allowed it to use the old "fiberpos" files and create a focalplane from that.  This was used to generate the `2010-01-01T00:00:00` model (the earliest date supported).  That resulting file was checked into svn, and was used to verify the use of desimodel focalplanes in fiberassign (comparing to the old hardcoded values).  We also began generating realistic focalplanes from the pos_settings svn repo.

3.  Later, a bug was discovered in the focalplane generation script.  When rotating petals from position "3" to the other positions, the positioner theta offset was not being rotated.  This was fixed, and now the positioner orientations were correctly rotated from petal-local coordinates to global focalplane coordinates.

4.  The legacy focalplane model was not modified after the fix in (3) and the svn version remains such that the theta offset is zero globally for all positioners across the focalplane.  This is exactly the model that was used in legacy fiberassign with hardcoded values.

5.  Meanwhile, the fiberassign code to compute positioner theta/phi angles to reach a specified X/Y location did not change.  The angle range-checking was only correct for the case of theta_offset == 0.  As the real positioner theta offset increases with petal location around the focalplane, a progressively larger "slice" of the patrol radius was being rejected as outside the allowed theta range.
